hal/armv7m/exceptions.c

```diff
@@ -29,6 +31,25 @@ void hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
 		"12 #Debug",      "13 #",        "14 #PendSV",    "15 #SysTick"
 	};
 	size_t i = 0;
+	u32 msp = (u32)ctx + sizeof(*ctx);
+	u32 psp = ctx->psp;
+	cpu_hwContext_t *hwctx;
+
+	/* If we came from userspace HW ctx in on psp stack */
+	if (ctx->excret == RET_THREAD_PSP) {
+		hwctx = (void *)ctx->psp;
+		msp -= sizeof(cpu_hwContext_t);
+		psp += sizeof(cpu_hwContext_t);
+#ifdef CPU_IMXRT /* FIXME - check if FPU was enabled instead */
+		psp += SIZE_FPUCTX;
```

Comment:
Right now - yes, it can be moved, but I would need to separate it again when lazy FPU context saving is implemented. I don't think putting it for now in `cpu_hwContext_t` would aid in debugging much, so I opt to leave this as is